components/ws-proxy/pkg/sshproxy/forward.go

```diff
@@ -53,32 +53,46 @@ func (s *Server) ChannelForward(ctx context.Context, session *Session, targetCon
 		close(maskedReqs)
 	}()
 
+	originChannelWg := sync.WaitGroup{}
+	originChannelWg.Add(3)
+	targetChannelWg := sync.WaitGroup{}
+	targetChannelWg.Add(3)
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+
 	go func() {
-		io.Copy(targetChan, originChan)
-		targetChan.CloseWrite()
+		defer wg.Done()
+		_, _ = io.Copy(targetChan, originChan)
+		targetChannelWg.Done()
+		targetChannelWg.Wait()
```

Comment:
I wonder should not we wait for stderr/stdout in both directions? won't close on one end trigger close on another

Reply:
yeah, I meant there we can simplify with one wg, which waits for both directions, but like that is fine as well 👍